File: src/test/java/com/dotnomi/fabricdependencyinjection/testclasses/identifier/DatabaseStorage.java
package com.dotnomi.fabricdependencyinjection.testclasses.identifier;

import com.dotnomi.fabricdependencyinjection.annotation.ModIdentifier;
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;

@ModScoped
@ModIdentifier("database")
public class DatabaseStorage implements StorageService {
  @Override
  public String store(String data) {
    return "Saved to database";
  }
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/identifier/DatabaseStorage.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.identifier

import com.dotnomi.fabricdependencyinjection.annotation.ModIdentifier
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped

@ModScoped
@ModIdentifier("database")
class DatabaseStorage : StorageService {
    override fun store(data: String): String {
        return "Saved to database"
    }
}
